src/contexts/ServerContext.js

```diff
@@ -91,36 +91,111 @@ export const ServerProvider = ({ children }) => {
-            method: 'GET',
-            url: `${restApiLocation}/query`,
-            headers: {
-                'Authorization': localStorage.getItem('zmt-token')
-            },
-            params: {
-                query_string: _query,
-                query_limit: limit,
-                row_offset: offset,
-                query_type: 'general'
+        let orderSyntax = order === 'asc' ? 1 : -1;
+        const resourceComparator = (a, b) => {
+            switch (orderBy) {
+                case 'RESC_TYPE_NAME':
+                    return orderSyntax * (a[1].localeCompare(b[1]))
+                case 'RESC_LOC':
+                    return orderSyntax * (a[4].localeCompare(b[4]))
+                case 'RESC_VAULT_PATH':
+                    return orderSyntax * (a[3].localeCompare(b[3]))
+                default:
+                    return orderSyntax * (a[0].localeCompare(b[0]))
             }
-        }).then((res) => {
-            setRescContext(res.data);
-            if (name === '') {
+        }
+        filteredResc.sort(resourceComparator);
+        filteredRescContext._embedded = filteredResc.slice(offset, offset + limit);
+        filteredRescContext.total = filteredResc.length;
+        setRescContext(filteredRescContext);
+    }
+
+    const loadResource = useCallback(async (offset, limit, name, order, orderBy) => {
+        setIsLoadingRescContext(true);
+        let base_query = `SELECT RESC_NAME,RESC_TYPE_NAME,RESC_ZONE_NAME,RESC_VAULT_PATH,RESC_LOC,RESC_INFO, RESC_FREE_SPACE, RESC_COMMENT,RESC_STATUS,RESC_CONTEXT,RESC_PARENT,RESC_ID WHERE RESC_NAME != 'bundleResc'`
+        if (name === '') {
+            let _query = queryGenerator(base_query, "asc", "RESC_NAME");
+            return axios({
+                method: 'GET',
+                url: `${restApiLocation}/query`,
+                headers: {
+                    'Authorization': localStorage.getItem('zmt-token')
+                },
+                params: {
+                    query_string: _query,
+                    query_limit: limit,
+                    row_offset: offset,
+                    query_type: 'general'
+                }
+            }).then((res) => {
+                setRescContext(res.data);
                 setRescTotal(res.data.total)
-            }
-            setIsLoadingRescContext(false);
-        }).catch(() => {
-            setRescContext(undefined)
-            setIsLoadingRescContext(false);
-        });
+                setIsLoadingRescContext(false);
+            }).catch(() => {
+                setRescContext(undefined)
+                setIsLoadingRescContext(false);
+            });
+        }
+        else {
+            let filteredResults = {};
+            let resc_name_query = `SELECT RESC_NAME,RESC_TYPE_NAME,RESC_ZONE_NAME,RESC_VAULT_PATH,RESC_LOC,RESC_INFO, RESC_FREE_SPACE, RESC_COMMENT,RESC_STATUS,RESC_CONTEXT,RESC_PARENT,RESC_ID WHERE RESC_NAME != 'bundleResc' AND RESC_NAME LIKE '%${name}%'`
```

Comment:
can we use base_query here and just add to it?   same a few lines lower... keep a single source of truth for this set of columns...